activesupport/lib/active_support/cache/redis_cache_store.rb

```diff
@@ -248,10 +248,11 @@ def delete_matched(matcher, options = nil)
             # Fetch keys in batches using SCAN to avoid blocking the Redis server.
             nodes = c.respond_to?(:nodes) ? c.nodes : [c]
 
-            nodes.each do |node|
+            nodes.each_with_object(Set.new) do |node, set|
               begin
                 cursor, keys = node.scan(cursor, match: pattern, count: SCAN_BATCH_SIZE)
                 node.del(*keys) unless keys.empty?
+                set.merge(keys)
```

Comment:
This set will consume memory proportional to the total number of entries deleted, where the current implementation is bounded by `SCAN_BATCH_SIZE`. It's also still implicitly coupled to the local cache; there's no other reason to return this data.

How about instead clearing the local cache entirely when `delete_matched` is called? More local cache entries will be removed than strictly necessary, but it's also a much simpler approach, and I think the edge case where it would matter (a single request reads some keys, deletes some other keys with a pattern, and then reads the original keys again) is less likely to be a problem than the increased memory usage described above.

This is what the local cache does for `cleanup`, which also deletes cache entries in a implementation-specific way:

https://github.com/rails/rails/blob/6a41940a965241ede9a77837169d6a329adf13d7/activesupport/lib/active_support/cache/strategy/local_cache.rb#L104-L108